### ISSUE TITLE
Master transition

### ISSUE THREAD

**Issue by guestisp** [lizardfs/lizardfs#693]
What happens by swapping the IP address between the current master and the current slave (and obviously, change the personality in config)

The slave become the new master, the old master become the new slave, but the newly-set slave (the "old master"), wuold be able to automatically connect to the new master and sync from that ?

**Comment by 4Dolio** [comment]
The old master should have been instructed to quickly stop and then a shadow promoted and then the IP moved. All chunks and clients and shadows and loggers will reconnect if/when their connections drop. The demoted master can be started again once it gives up the IP as a shadow then will then sync back up.

**Comment by guestisp** [comment]
Perfect. Thus, the older master (a transition could also be triggered manually) can automatically resync with the new master

**Comment by guestisp** [comment]
How did you tune the latency issues?

**Comment by 4Dolio** [comment]
I made my HA more patient when the master service 'appears' to be unresponsive.

**Comment by HerbertKnapp** [comment]
Yes, that works. Use case is power cuts which we have quite frequently.
I found existing free HA solutions cumbersome or requiring more than two machines and wrote a BASH script that runs on both Master and Shadow (1 Master, 1 Shadow, both identical) and it works like this:

If Shadow detects that Master is not reachable it changes its personality to Master and assigns itself the Master IP. When the former Master comes back, it detects that there's already a Master running and makes itself Shadow and assigns itself the Shadow IP, issues `mfsmetarestore -a` if necessary. 
Testing was done simply by randomly switching off power to either machine, often several times in a row. Gross latency (until client r/w resumes) is usually a few seconds up to a minute, mostly depending on how long it takes until all relevant chunkservers recognize the change (also see `MASTER_RECONNECTION_DELAY` and `MASTER_TIMEOUT`. Setting the latter to something lower than 1 caused instability even though ping is much lower than that.)
I tried to reduce this client latency by issuing arpings and similar methods to clear arp caches but couldn't reliably reproduce any improvements. It's probalby better left to the OS to handle this anyway.
Old Master machine, now a Shadow, then syncs with new Master. No problems there. Running a live VM off LFS while switching off Master is not recommended. It has worked in the past (with a very low load VM guest) but today, for the first time, it didn't and the VM crashed. 

Another more frequent use case is Master updates that require reboots. Procedure is logging in on Shadow first, making all necessary updates, rebooting it to see if there are any problems with the updates. If not, tell Shadow to reboot in say 15 minutes (or whenever, some time greater than it takes for Master updates, plus extra for boot time, etc.), then logging in Master, doing the updates and rebooting Master right after. This triggers the script that constantly checks for an available Master, switches Master and Shadow and 15 minutes later switches back again (not necessary since the machines are "identical" but I prefer it because of SSH key logins).

No lost chunks, done this dozens of times now either when updating or during testing phase with "pulling the plug" but I can't give any guarantees and would recommend doing this only when there is low load and no endangered chunks and when metadata versions of Master and Shadow are both the same.

Ideas for improvement much appreciated.
